### PR TITLE
[feat] integrated auth with profile context/onboarding

### LINKED ISSUE #54: Integrate Auth Context with Profile Context
Relevant PRs: 
* merged [ProfileContext](https://github.com/calblueprint/trap-garden/pull/52)
* original [ProfileContext PR](https://github.com/calblueprint/trap-garden/pull/24).
* AuthContext / AuthProvider file in repo.

To-do:
* remove hardcoded userId in /onboarding and ProfileProvider (use the auth.userId instead)
* Make sure that user's profile only appears if they're logged in, e.g. see [IJP](https://github.com/calblueprint/immigration-justice-project/blob/main/src/utils/ProfileProvider.tsx) 
* test if `hasPlot/setHasPlot` actually works (i.e. if you call setHasPlot in /onboarding, will it update elsewhere?)
* we might want to consider what to include in the dependency array for the useEffect in ProfileProvider that calls loadProfile (maybe we should reload profile every time auth.userId changes; otherwise we'd have to manually call loadProfile ourselves each time we need the most udpated `profile` value) 
* to test `hasPlot`, after the user submits the onboarding, they should be routed to `/view-plants`. If hasPlot is True, they the default viewing option in /view-plants should be myPlants; otherwise, it should be allPlants. (See the viewingOption state in `/view-plants`)
   * you'll need to call useProfile in /view-plants and access hasPlot! 

If you have time
* use the profile context in `/view-plants` to check that My Garden actually displays the plants for the signed in user, rather than the hardcoded userId (eventually we'd want to show an entirely different screen if the user is not signed in)

### PR DESCRIPTION
**Pull request by kevin3656**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
Integrated authcontext with profilecontext and onboarding, profile context will only receive profile if a user is logged in, onboarding will only upload if a user is logged in. User is redirected to view plants after onboarding with default view based on has_plot

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."

[image placeholder]: 

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'
files changed are mainly onboarding/page and utils/profileprovider who know use a userid based on auth instead of a hardcoded one. To test run app and login, then go through onboarding and supabase should receive a profile with the logged in users id. 

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
- Styling for onboarding !
- consider directly including the onboarding review page in the page container to avoid having to pass in so many additional props 

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'
- IJP's [ProfileProvider](https://github.com/calblueprint/immigration-justice-project/blob/main/src/utils/ProfileProvider.tsx)

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ccatherinetan

### CODE REVIEW

File: utils/ProfileProvider.tsx [hunk 1]
```diff
@@ -42,50 +40,64 @@ interface ProfileProviderProps {
...
-
-    setProfileData(fetchedProfile);
-    setProfileReady(true);
-  }, []);
+    try {
+      setProfileReady(false);
+      const fetchedProfile = await fetchProfileById(userId as UUID);
+      setProfileData(fetchedProfile);
+    } catch (error) {
+      console.error('Error fetching profile:', error);
```
Comment (ccatherinetan) [inline utils/ProfileProvider.tsx:58]: we might want to do error.message, i.e. 
` console.error('Error fetching profile:', error.message);`

File: utils/ProfileProvider.tsx [hunk 2]
```diff
@@ -42,50 +40,64 @@ interface ProfileProviderProps {
...
+    }
+  }, [authLoading, loadProfile]);
 
   const setProfile = useCallback(async (completeProfile: Profile) => {
     try {
       const updatedProfile = await upsertProfile(completeProfile);
       setProfileData(updatedProfile);
-      // Update has_plot if necessary by separate logic
     } catch (error) {
       console.error('Error setting profile:', error);
```
Comment (ccatherinetan) [inline utils/ProfileProvider.tsx:77]: let's use error.message?
also, we don't need to both console.error AND throw a new error. 
Since we're already catching an error from upsertProfile, I think we can just console.error here

File: utils/ProfileProvider.tsx [hunk 3]
```diff
@@ -42,50 +39,67 @@ interface ProfileProviderProps {
...
+      console.error('Error fetching profile:', error);
+    } finally {
+      setProfileReady(true);
+    }
+  }, [userId]);
 
   useEffect(() => {
-    loadProfile();
-  }, [loadProfile]);
+    if (!authLoading) {
```
Comment (ccatherinetan) [inline utils/ProfileProvider.tsx:65]: a possible edge case to consider is what if we want to call loadProfile (i.e. outside of this useEffect); perhaps we should handle the !authLaoding check inside loadProfile??

edit: the only place IJP used this was in /login, when deciding where to route the user. In the case of /login, we would already have access to authLoading, so we can leave it as is!

**Review by ccatherinetan — COMMENTED** [review]
looking great; left some remaining action items in the PR description, but will merge now!
